src/components/ImageSettingsModal.jsx

```diff
@@ -15,38 +15,62 @@ export default class ImageSettingsModal extends Component {
       newFileName: fileName,
       sha: '',
       content: null,
+      isPendingUpload: null,
     };
   }
 
   async componentDidMount() {
-    const { match, image: { fileName } } = this.props;
+    const { match, image, isPendingUpload } = this.props;
     const { siteName } = match.params;
-    const { data: { sha, content } } = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/images/${fileName}`, {
-      withCredentials: true,
-    });
-    this.setState({ sha, content });
+    if (isPendingUpload) {
+      const { content } = image;
+      this.setState({ content, isPendingUpload });
+    } else {
+      const { fileName } = image;
+      const { data: { sha, content } } = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/images/${fileName}`, {
+        withCredentials: true,
+      });
+      this.setState({ sha, content, isPendingUpload });
```

Comment:
No point setting state for `isPendingUpload`, since it has the same value of a prop with the same name